### PR TITLE
DOCS-2776: Move data dataset cli

### PR DESCRIPTION
**Pull request by sguequierre**
* add missing dataset commands (tested)
* move data dataset -> dataset or dataset data 
* add missing examples
* put named arguments in alphabetical order 
* clean up data section/add tags to examples

### CODE REVIEW

File: docs/cli.md [hunk 1]
```diff
@@ -286,17 +286,106 @@ viam organizations --help
...
+viam dataset list --dataset-ids=123,456
+
+# delete the specified dataset
+viam dataset delete --dataset-id=123
+
+# export dataset abc to output directory ./dataset/example in two folders called "data" and "metadata"
+viam dataset export --destination=./dataset/example --dataset-id=abc
+
+# add images tagged with the "example" tag between January and October of 2023 to dataset abc
+viam dataset data add filter --dataset-id=abc --location-id=123 --org-id=123 --start=2023-01-01T05:00:00.000Z --end=2023-10-01T04:00:00.000Z --tags=example
```
Comment (npentrel) [inline docs/cli.md:329]:
```suggestion
viam dataset data add filter --dataset-id=abc --location-id=123 --org-id=456 --start=2023-01-01T05:00:00.000Z --end=2023-10-01T04:00:00.000Z --tags=example
```
probably worth differentiating

File: docs/cli.md [hunk 2]
```diff
@@ -286,17 +286,106 @@ viam organizations --help
...
+
+# remove images tagged with the "example" tag between January and October of 2023 to dataset abc
+viam dataset data remove filter --dataset-id=abc --location-id=123 --org-id=123 --start=2023-01-01T05:00:00.000Z --end=2023-10-01T04:00:00.000Z --tags=example
```
Comment (npentrel) [inline docs/cli.md:332]:
```suggestion
viam dataset data remove filter --dataset-id=abc --location-id=123 --org-id=456 --start=2023-01-01T05:00:00.000Z --end=2023-10-01T04:00:00.000Z --tags=example
```

File: docs/cli.md [hunk 3]
```diff
@@ -286,17 +286,106 @@ viam organizations --help
...
+
+# add images with file IDs 123,456 in the org 123 and location 123 to dataset 123
+viam dataset data add ids --dataset-id=123 --org-id=123 --location-id=123 --file-ids=123,456
```
Comment (npentrel) [inline docs/cli.md:335]:
```suggestion
# add images with file IDs aaa and bbb in the org 123 and location 456 to dataset abc
viam dataset data add ids --dataset-id=abc --org-id=123 --location-id=456 --file-ids=aaa,bbb
```

File: docs/cli.md [hunk 4]
```diff
@@ -286,17 +286,106 @@ viam organizations --help
...
+
+# remove images with file IDs 123,456 in the org 123 and location 123 from dataset 123
+viam dataset data remove ids --dataset-id=123 --org-id=123 --location-id=123 --file-ids=123,456
```
Comment (npentrel) [inline docs/cli.md:338]:
```suggestion
# remove images with file IDs aaa and bbb in the org 123 and location 456 from dataset abc
viam dataset data remove ids --dataset-id= abc --org-id=123 --location-id= 456 --file-ids=aaa,bbb
```

File: docs/cli.md [hunk 5]
```diff
@@ -286,17 +286,106 @@ viam organizations --help
...
+#### Command options
+
+<!-- prettier-ignore -->
+| Command option | Description | Positional arguments |
+| -------------- | ----------- | -------------------- |
+| `create` | Create a new dataset | - |
+| `rename` | Rename an existing dataset | - |
+| `list` | List dataset information from specified IDs or for an org ID | - |
+| `delete` | Delete a dataset | - |
+| `data` | `add` or `remove` a new image to an existing dataset by its file id, or `add` or `remove` a group of images by specifying a filter | `ids`, `filter` |
```
Comment (npentrel) [inline docs/cli.md:350]:
```suggestion
| `data add` | Add a new image to an existing dataset by its file id, add a group of images by specifying a [filter](#using-the-filter-argument) | `ids`, `filter` |
| `data remove` | Remove a new image to an existing dataset by its file id, or remove a group of images by specifying a [filter](#using-the-filter-argument) | `ids`, `filter` |
```

File: docs/cli.md [hunk 6]
```diff
@@ -286,17 +286,106 @@ viam organizations --help
...
+| `--dataset-ids` | Dataset IDs of datasets to be listed. To retrieve these IDs, navigate to your dataset’s page in the [Viam app](https://app.viam.com), click **…** in the left-hand menu, and click **Copy dataset ID** | `list` | Optional |
+| `--destination` | Output directory for downloaded data | `export` | **Required** |
+| `--end` | ISO-8601 timestamp indicating the end of the interval | `data` | Optional |
+| `--file-ids` | File-ids to add or remove from a dataset | `data`, `export` | **Required** |
+| `--include-jsonl` | Set to `true` to include JSON Lines files for local testing |`export`| Optional |
+| `--location-id` | Location ID for the file ids being added or removed from the specified dataset (only accepts one location id) | `data` | **Required** |
+| `--name` | The name of the dataset to create or rename | `create`, `rename` | **Required** |
+| `--org-id` | Organization ID in which to edit the dataset | `create`, `data`, `list` | **Required** |
```
Comment (npentrel) [inline docs/cli.md:376]:
```suggestion
| `--org-id` | Organization ID of the organization the dataset belongs to | `create`, `data add`, `data delete`, `list` | **Required** |
```

File: docs/cli.md [hunk 7]
```diff
@@ -286,17 +286,106 @@ viam organizations --help
...
+##### Named arguments
+
+<!-- prettier-ignore -->
+| Argument | Description | Applicable commands | Required? |
+| -------- | ----------- | ------------------- | --------- |
+| `--dataset-id` | Dataset to add or remove images from, download, or rename. To retrieve these IDs, navigate to your dataset’s page in the [Viam app](https://app.viam.com), click **…** in the left-hand menu, and click **Copy dataset ID** | `rename`, `delete`, `data`, `export` | **Required** |
+| `--dataset-ids` | Dataset IDs of datasets to be listed. To retrieve these IDs, navigate to your dataset’s page in the [Viam app](https://app.viam.com), click **…** in the left-hand menu, and click **Copy dataset ID** | `list` | Optional |
+| `--destination` | Output directory for downloaded data | `export` | **Required** |
+| `--end` | ISO-8601 timestamp indicating the end of the interval | `data` | Optional |
+| `--file-ids` | File-ids to add or remove from a dataset | `data`, `export` | **Required** |
```
Comment (npentrel) [inline docs/cli.md:372]: This description doesn't make sense for an export command.

File: docs/cli.md [hunk 8]
```diff
@@ -286,17 +286,106 @@ viam organizations --help
...
+| `filter` | `add` or `delete` images from a dataset using a filter. See [Using the `filter` argument)](#using-the-filter-argument).|
+| `ids` | `add` or `delete` images from a dataset by specifying one or more file ids as a comma-separated list. See [Using the `ids` argument)](#using-the-ids-argument).|
+| `--help` | Return help |
+
+##### Named arguments
+
+<!-- prettier-ignore -->
+| Argument | Description | Applicable commands | Required? |
+| -------- | ----------- | ------------------- | --------- |
+| `--dataset-id` | Dataset to add or remove images from, download, or rename. To retrieve these IDs, navigate to your dataset’s page in the [Viam app](https://app.viam.com), click **…** in the left-hand menu, and click **Copy dataset ID** | `rename`, `delete`, `data`, `export` | **Required** |
```
Comment (npentrel) [inline docs/cli.md:368]:
```suggestion
| `--dataset-id` | Dataset to perform an operation on. To retrieve the ID, navigate to your dataset’s page in the [Viam app](https://app.viam.com), click **…** in the left-hand menu, and click **Copy dataset ID** | `rename`, `delete`, `data add`, `data delete`, `export` | **Required** |
```

File: docs/cli.md (hunk 7 above)
Comment (npentrel) [inline docs/cli.md:372]:
```suggestion
| `--file-ids` | The file-ids of the files to perform an operation on | `data add`, `data delete`, `export` | **Required** |
```

File: docs/cli.md [hunk 9]
```diff
@@ -286,17 +286,106 @@ viam organizations --help
...
+| `--dataset-ids` | Dataset IDs of datasets to be listed. To retrieve these IDs, navigate to your dataset’s page in the [Viam app](https://app.viam.com), click **…** in the left-hand menu, and click **Copy dataset ID** | `list` | Optional |
+| `--destination` | Output directory for downloaded data | `export` | **Required** |
+| `--end` | ISO-8601 timestamp indicating the end of the interval | `data` | Optional |
+| `--file-ids` | File-ids to add or remove from a dataset | `data`, `export` | **Required** |
+| `--include-jsonl` | Set to `true` to include JSON Lines files for local testing |`export`| Optional |
+| `--location-id` | Location ID for the file ids being added or removed from the specified dataset (only accepts one location id) | `data` | **Required** |
```
Comment (npentrel) [inline docs/cli.md:374]:
```suggestion
| `--location-id` | The location ID for the location in which to perform an operation (only accepts one location id) | `data add`, `data delete` | **Required** |
```

File: docs/cli.md [hunk 10]
```diff
@@ -377,39 +460,43 @@ done
...
+| `--parallel` | Number of download requests to make in parallel, with a default value of 10 |`export`, `delete`, `dataset export` | Optional |
 | `--part-id` | Filter by specified part ID |`export`, `delete`| Optional |
 | `--part-name` | Filter by specified part name |`export`, `delete`| Optional |
 | `--machine-id` | Filter by specified machine ID |`export`, `delete`, `tag filter` | Optional |
 | `--machine-name` | Filter by specified machine name |`export`, `delete`| Optional |
-| `--tags` | Filter by (`export`, `delete`, `dataset`) or add (`tag`) specified tag (accepts comma-separated list) |`export`, `delete`, `dataset`, `tag` | Optional |
+| `--tags` | Filter by (`export`, `delete`) or add (`tag`) specified tag (accepts comma-separated list) |`export`, `delete`, `tag` | Optional |
 | `--password` | Password for the database user being configured | `database configure` | **Required** |
 
 ##### Using the `ids` argument
```
Comment (npentrel) [inline docs/cli.md:486]: why is this still here? This should be with the dataset section

File: docs/cli.md [hunk 11]
```diff
@@ -429,16 +516,24 @@ See [Datasets](/services/data/dataset/#datasets) for more information.
 
 ##### Using the `filter` argument
```
Comment (npentrel) [inline docs/cli.md:517]: I think this should also go to dataset section. And you can link to there from data without duplicating.

File: docs/cli.md [hunk 12]
```diff
@@ -286,17 +286,106 @@ viam organizations --help
...
+
+##### Named arguments
+
+<!-- prettier-ignore -->
+| Argument | Description | Applicable commands | Required? |
+| -------- | ----------- | ------------------- | --------- |
+| `--dataset-id` | Dataset to add or remove images from, download, or rename. To retrieve these IDs, navigate to your dataset’s page in the [Viam app](https://app.viam.com), click **…** in the left-hand menu, and click **Copy dataset ID** | `rename`, `delete`, `data`, `export` | **Required** |
+| `--dataset-ids` | Dataset IDs of datasets to be listed. To retrieve these IDs, navigate to your dataset’s page in the [Viam app](https://app.viam.com), click **…** in the left-hand menu, and click **Copy dataset ID** | `list` | Optional |
+| `--destination` | Output directory for downloaded data | `export` | **Required** |
+| `--end` | ISO-8601 timestamp indicating the end of the interval | `data` | Optional |
```
Comment (npentrel) [inline docs/cli.md:371]:
```suggestion
| `--end` | ISO-8601 timestamp indicating the end of the interval | `data add`, `data delete` | Optional |
```

File: docs/cli.md [hunk 13]
```diff
@@ -286,17 +286,106 @@ viam organizations --help
...
+| `--file-ids` | File-ids to add or remove from a dataset | `data`, `export` | **Required** |
+| `--include-jsonl` | Set to `true` to include JSON Lines files for local testing |`export`| Optional |
+| `--location-id` | Location ID for the file ids being added or removed from the specified dataset (only accepts one location id) | `data` | **Required** |
+| `--name` | The name of the dataset to create or rename | `create`, `rename` | **Required** |
+| `--org-id` | Organization ID in which to edit the dataset | `create`, `data`, `list` | **Required** |
+| `--parallel` | Number of download requests to make in parallel, with a default value of 100 | `export` | Optional |
+| `--start` | ISO-8601 timestamp indicating the start of the interval | `data`| Optional |
+| `--tags` | Filter by specified tag (accepts comma-separated list) | `data` | Optional |
```
Comment (npentrel) [inline docs/cli.md:379]:
```suggestion
| `--start` | ISO-8601 timestamp indicating the start of the interval | `data add`, `data delete` | Optional |
| `--tags` | Filter by specified tag (accepts comma-separated list) | `data add`, `data delete` | Optional |
```

**Review by npentrel — CHANGES_REQUESTED** [review]

File: docs/cli.md (hunk 13 above)
Comment (sguequierre) [reply docs/cli.md:379]: ok I wasn't sure if I should separate out or not -- will make this change everywhere then

File: docs/cli.md (hunk 10 above)
Comment (sguequierre) [reply docs/cli.md:486]: Bc its used for data tags as well. Can be used for either

File: docs/cli.md (hunk 11 above)
Comment (sguequierre) [reply docs/cli.md:517]: it's used for both but I will move to dataset section

File: docs/cli.md [hunk 14]
```diff
@@ -286,17 +286,167 @@ viam organizations --help
...
+viam dataset data add filter --dataset-id=abc --location-id=123 --org-id=456 --start=2023-01-01T05:00:00.000Z --end=2023-10-01T04:00:00.000Z --tags=example
+
+# remove images tagged with the "example" tag between January and October of 2023 to dataset abc
+viam dataset data remove filter --dataset-id=abc --location-id=123 --org-id=456 --start=2023-01-01T05:00:00.000Z --end=2023-10-01T04:00:00.000Z --tags=example
+
+# add images with file IDs aaa and bbb in the org 123 and location 456 to dataset abc
+viam dataset data add ids --dataset-id=abc --org-id=123 --location-id=456 --file-ids=aaa,bbb
+
+# remove images with file IDs aaa and bbb in the org 123 and location 456 from dataset abc
+viam dataset data remove ids --dataset-id= abc --org-id=123 --location-id= 456 --file-ids=aaa,bbb
```
Comment (sguequierre) [inline docs/cli.md:338]:
```suggestion
viam dataset data remove ids --dataset-id= abc --org-id=123 --location-id=456 --file-ids=aaa,bbb
```

File: docs/cli.md [hunk 15]
```diff
@@ -286,17 +286,167 @@ viam organizations --help
...
+| `--file-ids` | The file-ids of the files to perform an operation on | `data add`, `data remove`, `export` | **Required** |
+| `--include-jsonl` | Set to `true` to include JSON Lines files for local testing |`export`| Optional |
+| `--location-id` | The location ID for the location in which to perform an operation (only accepts one location id) | `data add`, `data remove` | **Required** |
+| `--name` | The name of the dataset to create or rename | `create`, `rename` | **Required** |
+| `--org-id` | Organization ID of the organization the dataset belongs to | `create`, `data add`, `data remove`, `list` | **Required** |
+| `--parallel` | Number of download requests to make in parallel, with a default value of 100 | `export` | Optional |
+| `--start` | ISO-8601 timestamp indicating the start of the interval | `data add`, `data remove` | Optional |
+| `--tags` | Filter by specified tag (accepts comma-separated list) | `data add`, `data remove` | Optional |
+
+##### Using the `ids` argument
```
Comment (npentrel) [inline docs/cli.md:382]:
```suggestion
##### Using the `--file-ids` argument
```

**Review by npentrel — APPROVED** [review]
thanks - looks good!

**Pull request merged**